casetokens.php

```diff
@@ -268,7 +273,13 @@ function casetokens_civicrm_tokenvalues(&$values, $cids, $job = NULL, $tokens =
           ));
       }
     }
-
+    //fill client values
+    if (!empty($caseContact['values'])) {
+      $contacts['client'] = civicrm_api3('Contact', 'getsingle', [
+        'id' => array_values($caseContact['values'])[0]['contact_id.id'],
```

Comment:
We can just add `is_sequential => 1` to the `CaseContact` API call so that results will be returned sequentially and we can use `$caseContact['values'][0]['contact_id.id']` directly here.

Reply:
changed